### PR TITLE
Added ability to skip specific pull request

### PR DESCRIPTION
**Pull request by 50kudos**
By adding do_not_merge label to any pull request that doesn't need
to be merged and built into staging

### CODE REVIEW

File: lib/flash_flow/deploy.rb [hunk 1]
```diff
@@ -88,7 +88,10 @@ def merge_pull_requests
         if remote.nil?
           raise RuntimeError.new("No remote found for #{pull_request.head.repo.ssh_url}. Please run 'git remote add *your_remote_name* #{pull_request.head.repo.ssh_url}' and try again.")
         end
-        merge_or_rollback(remote, pull_request.head.ref, pull_request.number)
+
+        unless @github.has_label?(pull_request.number, @github.do_not_merge_label)
```
Comment (bradleyjucsc) [inline lib/flash_flow/deploy.rb]: Seems strange to have the Github class know about the `do_not_merge_label`, but to have to pass it in here. How about adding a `has_do_not_merge_label?` to the Github class?

File: lib/flash_flow/github.rb [hunk 2]
```diff
@@ -51,19 +52,19 @@ def pull_requests
...
     def remove_unmergeable_label(pull_request_number)
-      if has_unmergeable_label?(pull_request_number)
+      if has_label?(pull_request_number, unmergeable_label)
         octokit.remove_label(repo, pull_request_number, unmergeable_label)
       end
     end
 
     def add_unmergeable_label(pull_request_number)
-      unless has_unmergeable_label?(pull_request_number)
+      unless has_label?(pull_request_number, unmergeable_label)
```
Comment (bradleyjucsc) [inline lib/flash_flow/github.rb:60]: I like this little refactor :+1:

File: lib/flash_flow/config.rb [hunk 3]
```diff
@@ -54,6 +54,7 @@ def self.defaults
           branch_info_file: 'README.rdoc',
           locking_issue_id: nil,
           unmergeable_label: 'unmergeable',
+          do_not_merge_label: 'do not merge',
```
Comment (bradleyjucsc) [inline lib/flash_flow/config.rb:57]: Please add this to the tests for config.

File: lib/flash_flow/deploy.rb (hunk 1 above)
Comment (50kudos) [inline lib/flash_flow/deploy.rb]: `@github` already knew another label (`unmergeable`) and the `has_do_not_merge_label?` generalized with `has_unmergeable_label?` into `has_label?`

Comment (bradleyjucsc) [inline lib/flash_flow/deploy.rb]: I like the generalization of the `has_label?` method. But my point is really about the `do_not_merge_label` instance variable. It's never used in the class, the only time it's ever used it's just passed back in from deploy.rb. So you should either use it, or get rid of it from that class.

You could probably just do this if you want, and get rid of that variable from the class:
`unless @github.has_label?(pull_request.number, Config.configuration.do_not_merge_label)`

File: lib/flash_flow/deploy.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@ def initialize(opts={})
       @force = opts[:force]
 
       @cmd_runner = CmdRunner.new(opts.merge(logger: logger))
-      @github = Github.new(Config.configuration.repo, unmergeable_label: Config.configuration.unmergeable_label)
+      @github = Github.new(Config.configuration.repo, unmergeable_label: Config.configuration.unmergeable_label, do_not_merge_label: Config.configuration.do_not_merge_label)
```
Comment (bradleyjucsc) [inline lib/flash_flow/deploy.rb]: This change can be removed.

File: lib/flash_flow/github.rb [hunk 5]
```diff
@@ -51,19 +52,19 @@ def pull_requests
     end
 
     def remove_unmergeable_label(pull_request_number)
```
Comment (bradleyjucsc) [inline lib/flash_flow/github.rb]: This change can be removed.

File: lib/flash_flow/github.rb [hunk 6]
```diff
@@ -9,6 +9,7 @@ def initialize(repo, opts={})
       initialize_connection!
       @repo = repo
       @unmergeable_label = opts[:unmergeable_label] || 'unmergeable'
+      @do_not_merge_label = opts[:do_not_merge_label] || 'do not merge'
```
Comment (bradleyjucsc) [inline lib/flash_flow/github.rb]: This change can be removed.

**Pull request merged**